sherpa/astro/xspec/tests/test_xspec.py

```diff
@@ -611,3 +575,138 @@ def __init__(self, name='zbabs'):
     expected = XSzbabs()([1, 2, 3])
 
     assert_array_equal(expected, actual)
+
+
+# The following repeats logic in self.assertFinite above.
+# The plan is to move the SherpaTestCase derived tests into
+# plain pytest-style tests, and then self.assertFinite
+# will be removed, but that day is not today.
```

Comment:
If I am not mistaken `assertFinite` only uses `self` to use a specific implementation of the assertions, which I am assuming are equivalent to the `pytest`-powered assertions in `assert_is_finite`. Have you considered changing the implementation of `assertFinite` as just a call to `assert_is_finite`? Possibly making `assertFinite` a  static method?

Reply:
Well, long-term I want to get rid of `assertFinite`, but replacing it in the current code base would just tweak other tests (due to the imprecisely-defined meaning of the `model` argument) and I did not want to jump down that rabbit hole with this PR. I could do if you would rather it get cleaned up now (not a huge amount of work)?

Reply:
although, I did manage to bork my XSPEC 12.10.0 installation last night in tracking down some other installation, so would rather not make changes just at this time (as I can't easily validate them).